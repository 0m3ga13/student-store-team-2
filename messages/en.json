{
    "Index": {
        "Title": "MiniStore",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Free delivery": "Free delivery",
        "feature descriprion 1": "You can receive your order without incurring any additional shipping costs",
        "qualite guarantee": "qualite guarantee",
        "feature descriprion 2": "Our products or services meet high standards and perform as promised.",
        "daily offers": "daily offers",
        "feature descriprion 3": "Exclusive discounts and promotions on a daily basis",
        "100% secure payement": "100% secure payement",
        "feature descriprion 4": "Your financial information is protected and your transactions are safe from unauthorized access",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "my stuff":"my stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "electronics": "electronics",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product."
    }
}
