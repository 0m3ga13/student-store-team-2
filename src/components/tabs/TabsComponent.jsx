"use client";
import React, { useRef, useEffect, useState } from "react";

function TabsComponent({ items }) {
    const [selectedSection, setSelectedSection] = useState(0);
    const [selectedTitle, setSelectedTitle] = useState(0);
    const defaultBtnRef = useRef();

    useEffect(() => {
        if (defaultBtnRef.current) {
            defaultBtnRef.current.focus();
        }
    }, []);

    return (
        <div className='flex justify-center items-center py-8'>
            <div className='w-3/4 flex flex-col justify-start gap-y-6 '>
                <div className='flex items-center font-medium text-gray-700 text-sm border-b border-gray-200'>
                    {items
                        ? items.map((section, sectionIndex) =>
                              section.title.map((title, titleIndex) => (
                                  <button
                                      key={`${sectionIndex}.${titleIndex}`}
                                      ref={
                                          titleIndex === 0 && sectionIndex === 0
                                              ? defaultBtnRef
                                              : null
                                      }
                                      className='px-4 h-full rounded-sm uppercase text-[#55585B] hover:bg-[#89ceecba] hover:text-white hover:rounded-sm hover:text-base focus:drop-shadow-md focus:bg-[#72adc7ba] focus:text-white focus:rounded-sm focus:border-2 focus:border-gray-200 focus:text-semibold'
                                      onClick={() => {
                                          setSelectedSection(sectionIndex);
                                          setSelectedTitle(titleIndex);
                                      }}
                                  >
                                      {title}
                                  </button>
                              ))
                          )
                        : null}
                </div>
                <div className='bg-[#EDF1F3] p-1 m-2 rounded flex items-start text-sm font-base text-gray-700 '>
                    {items
                        ? items.map((item, index) => (
                              <div
                                  key={index}
                                  className={`${
                                      selectedSection === index
                                          ? "border-2 border-gray-200 rounded-lg p-4"
                                          : "hidden"
                                  }`}
                              >
                                  <h1 className='text-3xl text-[#72adc7]'>
                                      {item.content.title}
                                  </h1>
                                  <p className='text-[#3A3A3A] mt-8'>
                                      {item.content.text}
                                  </p>
                                  {selectedSection === index &&
                                      selectedTitle === 1 && (
                                          <div
                                              key={`${index + 1}`}
                                              className='border-2 border-gray-200 rounded-lg p-4'
                                          >
                                              {item.location}
                                          </div>
                                      )}
                              </div>
                          ))
                        : null}
                </div>
            </div>
        </div>
    );
}

export default TabsComponent;
