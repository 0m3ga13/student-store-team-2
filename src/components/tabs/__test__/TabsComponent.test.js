import TabsComponent from "../TabsComponent";
import { NextIntlClientProvider } from "next-intl";
import { messages } from "../../../../messages/en.json";
import renderer from "react-test-renderer";

it("renders correctly", () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <TabsComponent />
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
