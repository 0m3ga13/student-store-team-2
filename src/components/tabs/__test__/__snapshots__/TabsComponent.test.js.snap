// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex justify-center items-center py-8"
>
  <div
    className="w-3/4 flex flex-col justify-start gap-y-6 "
  >
    <div
      className="flex items-center font-medium text-gray-700 text-sm border-b border-gray-200"
    />
    <div
      className="bg-[#EDF1F3] p-1 m-2 rounded flex items-start text-sm font-base text-gray-700 "
    />
  </div>
</div>
`;
